controllers/storagecluster/storageconsumer.go

```diff
@@ -130,3 +108,132 @@ func (s *storageConsumer) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.St
 	// cleaned up via owner references
 	return ctrl.Result{}, nil
 }
+
+func getStorageClassesForDistribution(ctx context.Context, kubeClient client.Client, storageCluster *ocsv1.StorageCluster) (
+	[]ocsv1a1.StorageClassSpec, error) {
+
+	storageClassNames := map[string]bool{}
+	storageClassNames[util.GenerateNameForCephBlockPoolStorageClass(storageCluster)] = true
+	storageClassNames[util.GenerateNameForCephFilesystemStorageClass(storageCluster)] = true
+
+	crd := &metav1.PartialObjectMetadata{}
+	crd.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+	crd.Name = VirtualMachineCrdName
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(crd), crd); client.IgnoreNotFound(err) != nil {
+		return nil, err
+	}
+	if crd.UID != "" {
+		storageClassNames[util.GenerateNameForCephBlockPoolVirtualizationStorageClass(storageCluster)] = true
+	}
+
+	if storageCluster.Spec.ManagedResources.CephNonResilientPools.Enable {
+		storageClassNames[util.GenerateNameForNonResilientCephBlockPoolStorageClass(storageCluster)] = true
+	}
+
+	if storageCluster.Spec.NFS != nil && storageCluster.Spec.NFS.Enable {
+		storageClassNames[util.GenerateNameForCephNetworkFilesystemStorageClass(storageCluster)] = true
+	}
+
+	if storageCluster.Spec.Encryption.StorageClass && storageCluster.Spec.Encryption.KeyManagementService.Enable {
+		storageClassNames[util.GenerateNameForEncryptedCephBlockPoolStorageClass(storageCluster)] = true
+	}
+
+	// for day2 storageclasses
+	storageClassesInCluster := &storagev1.StorageClassList{}
+	if err := kubeClient.List(ctx, storageClassesInCluster); err != nil {
+		return nil, err
+	}
+	for idx := range storageClassesInCluster.Items {
+		// TODO: skip storageclasses that are from external mode if both internal & external mode is enabled
+		sc := &storageClassesInCluster.Items[idx]
+		scName := sc.Name
+		scProvisioner := sc.Provisioner
+		if slices.Contains(supportedCsiDrivers, scProvisioner) && !storageClassNames[scName] {
+			storageClassNames[scName] = true
+		}
+	}
+
+	scSpec := make([]ocsv1a1.StorageClassSpec, len(storageClassNames))
+	idx := 0
+	for scName := range maps.Keys(storageClassNames) {
+		// TODO: store the spec as the value which allows each value to be customizable corresponding to the name,
+		// not doing now as we only have name in the storageclassspec
+		scSpec[idx] = ocsv1a1.StorageClassSpec{Name: scName}
+		idx++
+	}
```

Comment:
As the wanted capacity is known there is no point in using an index. Using append is more idiomatic and with a known capacity will not re-allocate

```suggestion
	scSpec := make([]ocsv1a1.StorageClassSpec, 0, len(storageClassNames))
	for scName := range maps.Keys(storageClassNames) {
		// TODO: store the spec as the value which allows each value to be customizable corresponding to the name,
		// not doing now as we only have name in the storageclassspec
		scSpec = append(scSpec, ocsv1a1.StorageClassSpec{Name: scName})
	}
```

Reply:
done.